### PR TITLE
log-backup: update the endpoints when etcd cluster config changes

### LINKED ISSUE #14165: log-backup: PITR cannot detect new members in the etcd cluster
### What version of TiKV are you using?
v6.5.0

### What operating system and CPU are you using?
N/A, this problem is probably CPU irrelatived.

### Steps to reproduce
1. Run a cluster with PITR. (Assume PD nodes = `[1]`)
2. Scale-out the PD cluster. (Assume Added nodes `[2, 3]`)
3. Scale-in the PD cluster. (Assume expelling the nodes `[1, 2]`)

### What did you expect?
PITR work as normal.

### What did happened?
PITR failed to access the etcd cluster.

### PR DESCRIPTION
**Pull request by YuJuncen**
### What is changed and how it works?

Issue Number: Close #14165 

What's Changed:
This PR added a daemon thread to poll the current state of the etcd cluster members. So when PD cluster config changes, we can detect it and update the config.
This PR have also updated the version of `etcd-client`, which allows it to ignore some of PD nodes when they are failed. The change can be found at https://github.com/pingcap/etcd-client/pull/3.

```commit-message
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Manual test (add detailed scripts or steps below)

Side effects

- Performance regression
    - Consumes more CPU
    - Consumes more MEM

### Release note 

```release-note
Fixed a bug that may cause PiTR doesn't work when PD cluster config changed
```

### CODE REVIEW

File: components/backup-stream/src/metadata/store/etcd.rs [hunk 1]
```diff
@@ -301,3 +487,126 @@ impl Snapshot for EtcdSnapshot {
...
+                endpoints.push(url);
+            }
+            endpoints
+        }
+
+        fn add_member(&mut self) -> u64 {
+            let id = self.new_id();
+            let mut mem = PbMember::default();
+            mem.id = id;
+            mem.client_ur_ls = vec![format!("fakestore://{}", id)];
```
Comment (3pointer) [inline components/backup-stream/src/metadata/store/etcd.rs:554]: 🤔

File: components/backup-stream/src/metadata/store/lazy_etcd.rs [hunk 2]
```diff
@@ -203,3 +248,15 @@ impl MetaStore for LazyEtcdClient {
         self.get_cli().await?.txn_cond(txn).await
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::LazyEtcdClient;
+
+    #[test]
+    fn test_normalize_url() {
```
Comment (3pointer) [inline components/backup-stream/src/metadata/store/lazy_etcd.rs:257]: Do we need add more test cases here?

**Review by Leavrth — APPROVED** [review]

File: components/backup-stream/src/metadata/store/etcd.rs (hunk 1 above)
Comment (YuJuncen) [reply components/backup-stream/src/metadata/store/etcd.rs:554]: This URL won't be normalized. 🤔

**Review by 3pointer — APPROVED** [review]

**Review by hicqu — APPROVED** [review]

**Comment by YuJuncen** [comment]
(I don't think it needs rebase now...)

**Pull request merged**